lib/spree/localized_number.rb

```diff
@@ -37,6 +29,31 @@ def localize_number(*attributes)
       end
     end
 
+    def override_setter_for(attribute)
+      setter = "#{attribute}="
+      old_setter = instance_method(setter) if table_exists? && !column_names.include?(attribute.to_s)
+
+      define_method(setter) do |number|
+        validated_number = Spree::LocalizedNumber.validated_number(number)
+
+        if validated_number.nil?
+          @invalid_localized_number ||= []
+          @invalid_localized_number << attribute
+        end
+
+        if has_attribute?(attribute)
+          self[attribute] = validated_number
+        else
+          old_setter.bind(self).call(validated_number)
+        end
+      end
+    end
+
+    def self.validated_number(number)
+      return parse(number) if valid_localizable_number?(number)
+      nil
```

Comment:
I believe this nil would be implicitly returned by Ruby

Reply:
👍  that's true